File: ch22p751popovers/ch22p751popovers/ExtraViewController.m


#import "ExtraViewController.h"

@interface ExtraViewController ()

@end

@implementation ExtraViewController
- (IBAction)doButton:(id)sender {
    NSLog(@"extra view controller view frame: %@", NSStringFromCGRect(self.view.frame));
    [self dismissViewControllerAnimated:YES completion:^{
        NSLog(@"%@", @"dismissed");
    }];
}

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
        self.modalPresentationStyle = UIModalPresentationCurrentContext;
        self.preferredContentSize = CGSizeMake(320,220);
        // self.modalTransitionStyle = UIModalTransitionStyleFlipHorizontal;
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    // Do any additional setup after loading the view from its nib.
    self.view.frame = CGRectMake(0,0,320,220);
}


-(void)dealloc {
    NSLog(@"%@", @"dealloc extra view controller");
}

@end

File: iOS8bookExamples/bk2ch09p476popovers/ch22p751popovers/ExtraViewController.swift


import UIKit

// summoned by "i" button in first popover
// notice that preferred content size is obeyed

class ExtraViewController : UIViewController {
    
    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: NSBundle?) {
        super.init(nibName: "ExtraViewController", bundle: nibBundleOrNil)
        self.modalPresentationStyle = .CurrentContext
        self.preferredContentSize = CGSizeMake(320,220)
    }
    
    required init(coder: NSCoder) {
        fatalError("NSCoding not supported")
    }
    
    @IBAction func doButton(sender:AnyObject) {
        println("extra view controller view frame: \(self.view.frame)")
        self.dismissViewControllerAnimated(true, completion:{
            println("dismissed")
            })
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.view.frame = CGRectMake(0,0,320,220)
    }
    
    deinit {
        println("dealloc extra view controller")
    }
    
}